ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoiceTest.java

```diff
@@ -210,14 +225,31 @@ void onBlock_shouldReorgWhenProposerWeightingMakesForkBestChain() {
       expectedChainHead = chainBuilder.generateBlockAtSlot(currentSlot);
     }
 
-    importBlock(expectedChainHead);
-    assertThat(recentChainData.getStore().getProposerBoostRoot())
-        .contains(expectedChainHead.getRoot());
+    // advance time into slot
+    if (advanceTimeSlotMillis > 0) {
+      StoreTransaction transaction = recentChainData.startStoreTransaction();
+      UInt64 timeIntoSlotMillis =
+          recentChainData.getStore().getTimeMillis().plus(advanceTimeSlotMillis);
+      transaction.setTimeMillis(timeIntoSlotMillis);
+      transaction.commit().join();
```

Comment:
I'd use:
```suggestion
      SafeFutureAssert.safeJoin(transaction.commit());
```

otherwise it's possible for `.join()` to block indefinitely if the future doesn't complete which is incredibly hard to debug.

Reply:
cool, will change and keep in mind for the future